### ISSUE TITLE
Where roots are provided in description, add them as headwords.

### ISSUE THREAD

**Issue by vvasuki** [indic-dict/stardict-telugu#5]
From this [issue](https://github.com/sanskrit-coders/stardict-telugu/issues/3#issuecomment-195781908):

> So it is already providing not only base words, but description on how it evolved. there are codes in description if you see, like Skt, Tel, H, Tam,Drav,Port,Vik,Eng, and so on... examples are ....
> [Tadbava form of Skt. అప్సర.]
> [from Skt. మత్సరము.] , [Tel. fr. ఆ+చోటు.] , [Skt. from అగ్రము.] ,[Skt. from అతిథి], and so on... thesse are there for EVERY word.

It then makes sense to find strings like [from Skt. మత్సరము.] with headwords మాత్సర్యము|मात्सर्यमु|maatsaryamu, and do the following:
- remove telugu prathama-vibhakti suffixes for the indicated root words.
- add that word to the headwords list. So in this case, the new headword list would be: 
  మాత్సర్యము|మత్సరము|మత్సర|(the preceeding transliterated to devanAgarI and optitrans)

**Comment by damooo** [comment]
ॐ । 
It is major thing on todo list. but inside brackets not just words are there, but english description words, referances and so on , So not possible to just extract main words easily. and little bit of manual editing also to be done, so postponing it for a week, in that mean time i will work, and personally attend all words.

**Comment by vvasuki** [comment]
but dAmodara, I think it's possible to do this programmatically to a large extant. You just need to find the regular expression: \[from Skt. (.+).\] and use the matched characters.

**Comment by damooo** [comment]
ॐ । 
ha. but when i got like 'from this word *** ' like phrases are using, not all have 'from' only, it uses fr, and from is not always at begining.and discriptions are even like
 [From Skt. Not in use.] , (no word, but present  usage status,),
[Tel. Colloquial for అంతయును] (here no from at all, and descriptive words are there.
[Tel. from అందము.] (here from is at middle)
[From అంత and దనుక] (here no language mentioned, and two words are joined with eng propositions)
[Tel; plu, అంబలులు, అంబళులు or అంబళ్లు." Gen. అంబటి] , it is no where fixing in any regex,
[Skt. అ not, గ = move or walk.] , ( in this case useful explanation , but no regex possible.)
[Tadbava form of Skt.  అప్సర.] , (here clear root word  is there, but cannot accommadate to into regex

So for now any regex i tried are not actually saticifieng. because those descriptions are written , so that to give clear picture, according to word, and hence format changing rapidly. I think it will be good if we have  to provide some manual interfiarance , though it take some time.

**Comment by vvasuki** [comment]
Hmm - I suppose a series of regexes would shorten the work. There are 550 matches for the regex provided earlier..

**Comment by vvasuki** [comment]
This finds 2k+ matches: `\[[a-zA-Z =.]+([^a-zA-Z =]+).\] `

**Comment by damooo** [comment]
those wild cards simply giving all those matched in ,,bracketted lines. but we want only main words inside them. i.e remove junk in side brackets. if they all fallow regular format, then we can remove those 'regular junk'. but here they are not in that way. those lines extraction is already done by, just getting all lines containing lang codes, tel, Skt, Drav, Port, Vik, etc.. in side brackets and removing empty  brackets with only lang code and not having other word. .... but extraction of main word  is  main problem for us.

**Comment by vvasuki** [comment]
>> those wild cards simply giving all those matched in ,,bracketted lines. but we want only main words inside them. i.e remove junk in side brackets. 

Wrong, friend. That is what the () in the expressions capture. See for example [here](http://regexone.com/lesson/capturing_groups).

**Comment by damooo** [comment]
Oh, thanks then .don't know extraction is there in regex, i saw full line, and thought it will give full line with out knowing brackets are for there for extraction. I am an electronics student. So i am not fully familer with all options. So when ever i think it may not possible, correct me if solution is there. it will save my time.

**Comment by vvasuki** [comment]
This script has an example of [such extraction in python](https://github.com/sanskrit-coders/stardict-sanskrit/blob/master/sa-head/kalpadruma-sa/mUlam/make_simple_dict.py). It may be useful to understand that code fully - just to get your bearings in python regex. Then, it may be useful to checkin whatever scripts you're using so that I can point out problems/ better ways.

For similar reasons, I suspect that the catalyst files are not very useful.

**Comment by damooo** [comment]
ॐ । Ok. but with out catalyst files it will be difficult due to lot of fragmentation. if they are not there how can we differentiate between normal devanagari , and hrasva modified , or panchamas added words. And in case of english, it is further more difficult as there are so many differant versions like anusvara corrected, diirgha corrected, both corrected, and , pratyaya removed etc, so many cases. and branches. And not all words have all.only they have which has necissity. it vastly made it easy for us to do things, and retain what ever we want, and add what we wish to, organisingly..

**Comment by vvasuki** [comment]
By all means, we can keep the catalyst files if you feel the need. However, if we have a *script* that generates an entire set of headwords given the meaning line, then the catalyst files are redundant.

eg. be able to get "అ|अ|a" from
```
"—————————————————————————<br>దేవనాగరి : अ <br>ఉచ్ఛారణ : a,a<br>—————————————————————————<br><br>[Drav.]<br><br>The base of the demonstrative pronoun  అతడు, వాడు, అది."<br><br>
"
```
By the way this reminds me - the original headword must be present in the meaning line - will open a separate issue for this.
